`Development`: Minor code improvements in different areas

MoSchmidt opened the pull request:
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://confluence.aet.cit.tum.de/spaces/AP/pages/257786006/Language+Guidelines+Client).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://confluence.aet.cit.tum.de/spaces/AP/pages/256311714/PR+Guidelines).

#### Server

- [x] **Important**: I implemented the changes with a [very good performance](https://confluence.aet.cit.tum.de/spaces/AP/pages/256311718/Performance+Guidelines) and prevented too many (unnecessary) and too complex database calls.
- [x] I **strictly** followed the principle of **data economy** for all database calls.
- [x] I **strictly** followed the [server coding and design guidelines](https://confluence.aet.cit.tum.de/spaces/AP/pages/256311877/Server+Guidelines).
- [x] I documented the Java code using JavaDoc style.

#### Client

- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data (e.g. using paging).
- [x] I **strictly** followed the principle of **data economy** for all client-server REST calls.
- [x] I **strictly** followed the [client coding and design guidelines](https://confluence.aet.cit.tum.de/spaces/AP/pages/256311716/Client+Guidelines).
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://confluence.aet.cit.tum.de/spaces/AP/pages/256311925/Client+Test+Guidelines).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context

Improved the evaluation module and fixed minor issues.

### Description

- Renamed the professor sidebar items for "Applications"
- added sidebar icons for the two evaluation pages
- Adapted the overview page to translation
- Fixed Styling issues
- Added HEaders for the evaluation pages

### Steps for Testing

- [ ] Check the functionality of the evaluation pages
- [ ] Verify the styling of the evaluation pages


Prerequisites:

1. Log in as a Professor
2. Navigate to the "Overview" and "Review" screen

### Review Progress

#### Code Review

- [ ] Code Review 1

#### Manual Tests

- [ ] Test 1


### Screenshots

<img width="2557" height="1005" alt="grafik" src="https://github.com/user-attachments/assets/ff6146b6-b7ac-4cc3-8ab1-842713d092ff" />

<img width="2559" height="1126" alt="grafik" src="https://github.com/user-attachments/assets/9a653fd1-29e3-4cad-bbb7-acdaf95424fb" />

Douglas-Sarah reviewed (changes requested):
Functionality and styling LGTM, I have just left a small comment.

I have an additional question- would it be possible to add translation here as well? It is not crucial for release, but in this PR you already added some other translations for the evaluation page, so I thought it was fitting:
<img width="779" height="374" alt="image" src="https://github.com/user-attachments/assets/15d2cb3d-cba5-40ab-93fc-90d738d480f9" />

Also, when the professor opens an unopened letter, it should change its state to InReview, right? Because at the moment, that does not yet happen.
<img width="1211" height="868" alt="image" src="https://github.com/user-attachments/assets/040695be-a2af-4df3-a4aa-ae914f75b826" />

MoSchmidt commented:
> Functionality and styling LGTM, I have just left a small comment.
> 
> I have an additional question- would it be possible to add translation here as well? It is not crucial for release, but in this PR you already added some other translations for the evaluation page, so I thought it was fitting: <img alt="image" width="779" height="374" src="https://private-user-images.githubusercontent.com/198627919/465368701-15d2cb3d-cba5-40ab-93fc-90d738d480f9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTIyNTEzNzYsIm5iZiI6MTc1MjI1MTA3NiwicGF0aCI6Ii8xOTg2Mjc5MTkvNDY1MzY4NzAxLTE1ZDJjYjNkLWNiYTUtNDBhYi05M2ZjLTkwZDczOGQ0ODBmOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNzExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDcxMVQxNjI0MzZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wNjFlMDZjOGExMDVkMTliODI0ZTRhZjljMTIyMDhlZGJlYjJjZmFjOWIyMWRlZmE0MzMyNDViM2ZhY2QzOGU1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.G24Kjv7UGD0ja7muz3jJov291Mzrxju0SpFyt7t2Q58">
> 
> Also, when the professor opens an unopened letter, it should change its state to InReview, right? Because at the moment, that does not yet happen. <img alt="image" width="1211" height="868" src="https://private-user-images.githubusercontent.com/198627919/465369535-040695be-a2af-4df3-a4aa-ae914f75b826.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTIyNTEzNzYsIm5iZiI6MTc1MjI1MTA3NiwicGF0aCI6Ii8xOTg2Mjc5MTkvNDY1MzY5NTM1LTA0MDY5NWJlLWEyYWYtNGRmMy1hNGFhLWFlOTE0Zjc1YjgyNi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNzExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDcxMVQxNjI0MzZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yMDMxYjk3NmJmOWRmNGIyMTAzYmQ5MzAzYmQzZTZmNWNjNjU5NjA1OTdkZGIzOTc3NTZlMzlmMWJhOWZiMDI4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.pBv_Dann4jwq0NiS59_RYy4nOTowQOmmVKMqT7RKQd8">

Thanks for the feedback. To add translation for the dropdown, we would need to adjust this component as it can not be handled from a parent component. This will be implemented in a future PR.

I have also now added the functionality that when opening a SENT (e.g., UNOPENED) application, its status will be updated. Therefore, I have also added a new endpoint and sevrver logic

MoSchmidt commented:
I have now also fixed the routing between the two evaluation screens

marc-fett commented on src/main/java/de/tum/cit/aet/evaluation/web/ApplicationEvaluationResource.java:
```diff
@@ -141,9 +122,41 @@ public ResponseEntity<ApplicationEvaluationDetailListDTO> getApplicationsDetails
+     *
+     * @return the current {@link User}
+     */
+    private User getCurrentUser() {
```
please move this to CurrentUserService

marc-fett commented on src/main/java/de/tum/cit/aet/evaluation/web/ApplicationEvaluationResource.java:
```diff
@@ -141,9 +122,41 @@ public ResponseEntity<ApplicationEvaluationDetailListDTO> getApplicationsDetails
+     * @return the research group ID
+     * @throws RuntimeException if no research group is associated with the user
+     */
+    private UUID getCurrentUserResearchGroup() {
```
please change the getResearchGroupIdIfProfessor in CurrentUserService instead

marc-fett reviewed (changes requested):
Nice work! Added some small comments and questions

MoSchmidt commented:
> Nice work! Added some small comments and questions

Thanks for the feedback. I have moved all current user-related logic to the `CurrentUserService`

MoSchmidt commented on src/main/webapp/i18n/de.json:
```diff
@@ -0,0 +1 @@
+{"activate":{"title":"Aktivierung","messages":{"success":"<strong>Ihr Benutzer wurde aktiviert.</strong> Bitte ","error":"<strong>Ihr Benutzer konnte nicht aktiviert werden.</strong> Bitte benutzen Sie die Registrierungsmaske, um sich zu registrieren."}},"tumApplyApp":{"adminAuthority":{"home":{"title":"Authorities","refreshListLabel":"Liste aktualisieren","createLabel":"Authority erstellen","cre ...
```
I removed all of its content. But I thought we should not include tehse files in the gitignore?

marc-fett replied:
I think Aniruddh changed his opion about it. But we need to remove it once anyways again

marc-fett reviewed (approved):
LGTM

az108 reviewed (approved):
LGTM
